test/unit/write_concern.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expect } from 'chai';
+
+import { WriteConcern } from '../mongodb';
+
+describe('WriteConcern', function () {
+  describe('#constructor', function () {
+    context('when w is provided', function () {
```

Comment:
Can we add a case for w being a string of digits `'234'` should work b/c of the `!Number.isNaN(Number(w))` case we have in the ctor

I guess we should also at least make sure majority works, but I think the intention is that any string could be provided if other `w` settings were introduced in the future.

Reply:
Add tests for string number and majority.